content/docs/guides/author_develop/pwa-amp/amp-in-pwa.md

```diff
@@ -0,0 +1,101 @@
+
+## Use the Shadow API to render a page inline
+
+Finally, when you want to display content after a user action, it's time to fetch the relevant AMP document and let Shadow AMP take over. First, implement a function to fetch the page, similar to this one:
+
+[sourcecode:javascript]
+function fetchDocument(url) {
+
+  // unfortunately fetch() does not support retrieving documents,
+  // so we have to resort to good old XMLHttpRequest.
+  var xhr = new XMLHttpRequest();
+
+  return new Promise(function(resolve, reject) {
+    xhr.open('GET', url, true);
+    xhr.responseType = 'document';
+    xhr.setRequestHeader('Accept', 'text/html');
+    xhr.onload = function() {
+      // .responseXML contains a ready-to-use Document object
+      resolve(xhr.responseXML);
+    };
+    xhr.send();
+  });
+}
+[/sourcecode]
+
+{% call callout('Important', type='caution') %}
+In order to simplify the above code example, we skipped over error handling. You should always make sure to catch and handle errors gracefully.
+{% endcall %}
+
+Now that we have our ready-to-use `Document` object, it's time to let AMP take over and render it. Get a reference to the DOM element that serves as container for the AMP document, then call `AMP.attachShadowDoc()`, like so:
+
+[sourcecode:javascript]
+// This can be any DOM element
+var container = document.getElementById('#container');
+
+// The AMP page you want to display
+var url = "https://my-domain/amp/an-article.html";
+
+// Use our fetchDocument method to get the doc
+fetchDocument(url).then(function(doc) {
+  // Let AMP take over and render the page
+  var ampedDoc = AMP.attachShadowDoc(container, doc, url);
+});
+[/sourcecode]
+
+{% call callout('Tip', type='success') %}
+Before you actually hand the document over to AMP, it's the perfect time to remove page elements that make sense when displaying the AMP page standalone, but not in embedded mode: For example, footers and headers.
+{% endcall %}
+
+And that's it! You should see the AMP page in all its glory, rendered as a child of your overall Progressive Web App.
+
+## Clean up after yourself
+
+Chances are your user will navigate from AMP to AMP within your Progressive Web App. When discarding the previous rendered AMP Page, always make sure to tell AMP about it, like so:
+
+[sourcecode:javascript]
+// ampedDoc is the reference returned from AMP.attachShadowDoc
+ampedDoc.close();
+[/sourcecode]
+
+This will tell AMP that you're not using this document any longer and will free up memory and CPU overhead.
+
+## See it in action
+
+{{ video('/static/img/docs/pwamp_react_demo.mp4', 620, 1100, align='right third', loop='true', autoplay='true', controls='') }}
+
+You can see the "AMP in PWA" pattern in action in the [React sample](https://github.com/ampproject/amp-publisher-sample/tree/master/amp-pwa) we've built. It demonstrates smooth transitions during navigation and comes with a simple React component that wraps the above steps.
```

Comment:
Might be worth explicitly calling out the "best of both worlds" situation by using custom JS in the PWA shell and AMP as content. 

The React component can be used without the sample and is published on npm here: https://www.npmjs.com/package/react-amp-document